src/notebooks/execution/cellExecutionMessageHandlerFactory.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { Kernel } from '@jupyterlab/services';
+import { NotebookCell, NotebookController } from 'vscode';
+import { ITracebackFormatter } from '../../kernels/types';
+import { IApplicationShell } from '../../platform/common/application/types';
+import { disposeAllDisposables } from '../../platform/common/helpers';
+import { IDisposable, IExtensionContext } from '../../platform/common/types';
+import { CellOutputDisplayIdTracker } from './cellDisplayIdTracker';
+import { CellExecutionMessageHandler } from './cellExecutionMessageHandler';
+
+export class CellExecutionMessageHandlerFactory {
```

Comment:
We have a 1-1 relationship between the IKernel and this class.
I.e. if we change the kernel, then this class dies and we end up with a whole new factory.

Reply:
Do you mean the controller? It doesn't seem to point to the kernel.

Reply:
sorry yes